addons/mail/models/mail_thread.py

```diff
@@ -2492,22 +2492,21 @@ def _notify_encode_link(self, base_link, params):
         return hm
 
     def _notify_get_action_link(self, link_type, **kwargs):
-        local_kwargs = dict(kwargs)  # do not modify in-place, modify copy instead
-        base_params = {
+        """ Prepare link to an action: view document, follow document, ... """
+        params = {
             'model': kwargs.get('model', self._name),
             'res_id': kwargs.get('res_id', self.ids and self.ids[0] or False),
         }
-
-        local_kwargs.pop('message_id', None)
-        local_kwargs.pop('model', None)
-        local_kwargs.pop('res_id', None)
+        params.update(dict(
+            (key, value)
+            for key, value in kwargs.items()
+            if key in ('access_token', 'auth_signup_token', 'auth_login')
```

Comment:
2 things here:

1. Is it ok for stable?
Previous implementation was pretty much blacklisting. Here you are whitelisting instead.
Is there any "security" based reasons to do so?
Because doing that in stable may break custom module implementations that rely on more values being passed to `_notify_encode_link`.
It's an edge case, but it would be safer to stick to previous implementation unless it causes issues?

2. If you end up keeping it, a small test ensuring those specific values are correctly passed and used would be nice.

Reply:
Hello,

1/ indeed we restrict what is accepted to make links. It is safer to white list than blacklist as now you may end up with a lot of noise (body, email_from, subject, ... ) . We cannot stick to previous implementation as we basically change how this method behaves. Anyway adding parameters to odoo url is not a safe way to do custom behavior as it is completely not supported outside of some model / res_id / action + auth parameters.

2/ You are doing too much JS.

Reply:
Sorry for too much JS. To make you happy again I wrote a test. Don't be sad. Otherwise I am sad. And when I am sad I picole and crac.

Reply:
Don't be, I'm expecting another "Happy TDE works with happy AWA" for my eval this year :D